pkg/util/profiling/profiling.go

```diff
@@ -48,7 +49,20 @@ func Start(apiKey, site, env, service string, period time.Duration, cpuDuration
 		profiler.WithProfileTypes(profiler.CPUProfile, profiler.HeapProfile, profiler.MutexProfile),
```

Comment:
Do we need to also enable the relevant profile types here to actually enable them in the DD profiler? https://docs.datadoghq.com/tracing/profiler/enabling/#pagetitle

Reply:
additional question for @Gui774ume : `profiler.MutexProfile` was added by default in https://github.com/DataDog/datadog-agent/pull/8025, but since it's documented as having additional overhead, wouldn't it be better to only enable it if an additional config option is enabled? (which is the approach of the present PR for the core agent)

Not urgent for 7.29 since internal profiling is an internal feature, but for 7.30.

Reply:
> Do we need to also enable the relevant profile types here to actually enable them in the DD profiler?
No, they will be enabled by passing `BlockProfileRate` and `MutexProfileFraction` options.